src/profile-logic/cpu.js

```diff
@@ -37,15 +37,33 @@ function _computeMaxVariableCPUCyclesPerMs(threads: RawThread[]): number {
 
     // Ignore the first CPU delta value; it's meaningless because there is no
     // previous sample.
-    for (let i = 1; i < samples.length; i++) {
-      const sampleTimeDeltaInMs = samples.time[i] - samples.time[i - 1];
-      if (sampleTimeDeltaInMs !== 0) {
-        const cpuDeltaPerMs = (threadCPUDelta[i] || 0) / sampleTimeDeltaInMs;
-        maxThreadCPUDeltaPerMs = Math.max(
-          maxThreadCPUDeltaPerMs,
-          cpuDeltaPerMs
-        );
+    const { time: samplesTimeCol, timeDeltas: samplesTimeDeltasCol } = samples;
+    if (samplesTimeCol !== undefined) {
+      for (let i = 1; i < samples.length; i++) {
+        const sampleTimeDeltaInMs = samplesTimeCol[i] - samplesTimeCol[i - 1];
+        if (sampleTimeDeltaInMs !== 0) {
+          const cpuDeltaPerMs = (threadCPUDelta[i] || 0) / sampleTimeDeltaInMs;
+          maxThreadCPUDeltaPerMs = Math.max(
+            maxThreadCPUDeltaPerMs,
+            cpuDeltaPerMs
+          );
+        }
       }
+    } else if (samplesTimeDeltasCol !== undefined) {
+      for (let i = 1; i < samples.length; i++) {
+        const sampleTimeDeltaInMs = samplesTimeDeltasCol[i];
```

Comment:
The only difference is how we gather the delta, it might be worth either extracting this part into a function (the only different part for the 2 cases) or computing the diffs first (in that case we'd loop twice in the first case, but maybe that's OK?).

I always find it hard when I look at similar-but-subtly-different pieces of code to actually tell what's different.

If you think that would be too complex for what we need, alternatively you can add comments on top of the different lines.

Reply:
Good point, I'll change it to compute the deltas first and then have shared code for the loop.

Reply:
Changed to:

```js

    const timeDeltas =
      samples.time !== undefined
        ? timeColumnToTimeDeltas(samples.time)
        : ensureExists(samples.timeDeltas);

    // Ignore the first CPU delta value; it's meaningless because there is no
    // previous sample.
    for (let i = 1; i < samples.length; i++) {
      const sampleTimeDeltaInMs = timeDeltas[i];
      if (sampleTimeDeltaInMs !== 0) {
        const cpuDeltaPerMs = (threadCPUDelta[i] || 0) / sampleTimeDeltaInMs;
        maxThreadCPUDeltaPerMs = Math.max(
          maxThreadCPUDeltaPerMs,
          cpuDeltaPerMs
        );
      }
    }
```